src/app_config.cpp

```diff
@@ -267,6 +269,44 @@ config_load_settings()
   }
 }
 
+void append_common_config(DynamicJsonDocument &doc) {
+  doc["config_version"] = config_version;
```

Comment:
moved `config_version` into version object for practical reasons when reading the mqtt topic and also to ensure the attributes of the mqtt mapping in HA gets updated correctly when we use one sensor for all attributes.

Reply:
config_version is already published in /status ( like all xxx_version props )  , or in main mqtt topic ( it's not part of the /config as it's not saved in the flash and is just here to track live updates. I don't think it should stays in the published config.

Reply:
I don't know HASS mqtt implementation, but it should clearly parse the Json when the topic is updated. Why does it has to check a specific property ?

Reply:
It parses it. When a mqtt update happens, a component can be bounded to a topic, a apecific field inside, and all other fields are attributes of the same sensor.

So in HA we can have only 1 sensor listener to 1 mqqtt topic and it will have its attributes taken from the payload.

But still ,there are needs to have at least one field representing this sensor. In the case of the config, this is the config version.

Each time the config version is changing, then the config attributes change also.

It is possible to have another topic defined for the attributes but it would need to be published BEFORE config_version is published. Let me try...

Reply:
I've also changed the ordering of the calls to `append_common_config(doc);`